### DISCUSSION TITLE
Installation in mac

### DISCUSSION THREAD

**Discussion by jaffa4321** [EtienneAb3d/karaok-AI#5] [Q&A]
i am getting this error while installing in my mac m1:

Java version : 11.0.13
JRE location : /Library/Java/JavaVirtualMachines/jdk-11.0.13.jdk/Contents/Home
SWT version : 4940
***WARNING: Display must be created on main thread due to Cocoa restrictions. Use vmarg -XstartOnFirstThread
org.eclipse.swt.SWTException: Invalid thread access
	at org.eclipse.swt.SWT.error(SWT.java:4869)
	at org.eclipse.swt.SWT.error(SWT.java:4784)
	at org.eclipse.swt.SWT.error(SWT.java:4755)
	at org.eclipse.swt.widgets.Display.error(Display.java:1103)
	at org.eclipse.swt.widgets.Display.createDisplay(Display.java:851)
	at org.eclipse.swt.widgets.Display.create(Display.java:835)
	at org.eclipse.swt.graphics.Device.<init>(Device.java:130)
	at org.eclipse.swt.widgets.Display.<init>(Display.java:734)
	at org.eclipse.swt.widgets.Display.<init>(Display.java:725)
	at com.cubaix.kaiDJ.KaiDJ.createInterface(KaiDJ.java:233)
	at com.cubaix.kaiDJ.KaiDJ.start(KaiDJ.java:1004)
	at com.cubaix.kaiDJ.KaiDJ.main(KaiDJ.java:1211)
	
Please tell me a solution to this Thankyou..!!

**Comment by EtienneAb3d** [comment]
@jaffa4321 
Can you try with this updated start shell?
(unzip the content, and replace the original `kaiDJ.sh` file)
[kaiDJ.sh.zip](https://github.com/EtienneAb3d/karaok-AI/files/12328913/kaiDJ.sh.zip)

**Reply by jaffa4321** [reply]
<img width="1427" alt="image" src="https://github.com/EtienneAb3d/karaok-AI/assets/121224007/b193d2d1-bb9c-48b6-bdfe-d9cee7aeca51">

getting like this :

Java version : 11.0.13
JRE location : /Library/Java/JavaVirtualMachines/jdk-11.0.13.jdk/Contents/Home
SWT version : 4940
MIXER not added : Port MacBook Air Microphone
MIXER not added : Port MacBook Air Speakers
MIXER not added : Port Microsoft Teams Audio
MIXER added : Default Audio Device
MIXER not added : MacBook Air Microphone
MIXER added : MacBook Air Speakers
MIXER added : Microsoft Teams Audio
Can't load file '/Users/balajikotni/karaok-AI/config.xml'
Transactions are supported.
Transactions are supported.
CAT table found.
MP3 table found.
DB loaded in : 00:00:61
Db cleaned in : 00:00:00
Can't load file '/Users/balajikotni/karaok-AI/last.jdj'
Last PL loaded in : 00:00:00
Using char encoding 'utf-8' for file names in zip file : /Users/balajikotni/Documents/bin/kaiDJ-Mac/demo/Katy Perry - Firework.mp3.kai
Unzipping : /Users/balajikotni/Documents/bin/kaiDJ-Mac/demo/Katy Perry - Firework.mp3.bass
Unzipping : /Users/balajikotni/Documents/bin/kaiDJ-Mac/demo/Katy Perry - Firework.mp3.drums
Unzipping : /Users/balajikotni/Documents/bin/kaiDJ-Mac/demo/Katy Perry - Firework.mp3.ksrt
Unzipping : /Users/balajikotni/Documents/bin/kaiDJ-Mac/demo/Katy Perry - Firework.mp3.other
Unzipping : /Users/balajikotni/Documents/bin/kaiDJ-Mac/demo/Katy Perry - Firework.mp3.vocals

**Comment by EtienneAb3d** [comment]
@jaffa4321 
It looks ok to start. But, the interface is not properly displayed. It will become hard to investigate without a direct access to a machine with the same behaviour.
:(

**Comment by EtienneAb3d** [comment]
@jaffa4321 
Can you try to download the "SWT Binary and Source" for your machine at the bottom of this page, and update the `swt-mac.jar` file with the `swt.jar` found in it:
https://download.eclipse.org/eclipse/downloads/drops4/S-4.29M1-202307051800/
